rclcpp/include/rclcpp/intra_process_manager_impl.hpp

```diff
@@ -265,11 +265,15 @@ class IntraProcessManagerImpl : public IntraProcessManagerImplBase
       return std::strcmp(lhs, rhs) < 0;
     }
   };
+
+  // Make sure allocator given to map has the correct value_type
+  using IDTopic = std::pair<const char *, AllocSet>;
+
   using IDTopicMap = std::map<
-      const char *,
-      AllocSet,
+      typename IDTopic::first_type,
+      typename IDTopic::second_type,
       strcmp_wrapper,
-      RebindAlloc<std::pair<const std::string, AllocSet>>>;
+      RebindAlloc<IDTopic>>;
```

Comment:
Is the use of the alias (i.e. the `using` statement) required or is it only necessary to change either the map declaration to `std::string` or the `RebindAlloc` declaration to `const char *`?

Reply:
I don't think it should be necessary, and this is evidence against this fix working, but I managed to get a CI job to run with that change dbb462b5bcad8ddaeb01468f2cd0fcf07cf53e65 and it encountered the same issue.

Reply:
Well, either way what you've put here seems fine, though I don't know why `const std::string` or `const char *` is preferable.

Reply:
Looks like the first type in the `std::pair` needs to be type `const Key`, which is `const char * const` (thanks grey). I ran a ci_osx and it successfully built rclcpp. If everything builds using ci_launcher then I'll squash/merge this.